pkg/util/helper/workstatus.go

```diff
@@ -308,3 +311,38 @@ func BuildStatusRawExtension(status interface{}) (*runtime.RawExtension, error)
 		Raw: statusJSON,
 	}, nil
 }
+
+// BuildPreservedLabelState builds preserved label state from the state preservation rules and raw status.
+func BuildPreservedLabelState(statePreservation *policyv1alpha1.StatePreservation, rawStatus []byte) (map[string]string, error) {
```

Comment:
Could we consider moving this parameter to the `binding.go` file or a new `failover.go` file?